plugin/evm/gossiper_orders.go

```diff
@@ -0,0 +1,182 @@
+				}
+			case orders := <-n.ordersToGossipChan:
+				for _, order := range orders {
+					orderHash, err := order.Hash()
+					if err != nil {
+						log.Error("failed to hash order", "err", err)
+						continue
+					}
+					n.ordersToGossip[orderHash] = order
+				}
+				if attempted, err := n.gossipSignedOrders(); err != nil {
+					log.Warn(
+						"failed to send signed orders",
+						"len(orders)", attempted,
+						"err", err,
+					)
+				}
+			case <-n.shutdownChan:
+				return
+			}
+		}
+	}, "panic in awaitSignedOrderGossip", orderbook.AwaitSignedOrdersGossipPanicsCounter)
+
+}
+
+func (n *pushGossiper) gossipSignedOrders() (int, error) {
+	//
+	if (time.Since(n.lastOrdersGossiped) < minGossipOrdersBatchInterval) || len(n.ordersToGossip) == 0 {
+		return 0, nil
+	}
+	n.lastOrdersGossiped = time.Now()
+	orders := []*hubbleutils.SignedOrder{}
+	for orderHash, order := range n.ordersToGossip {
+		if len(orders) >= maxSignedOrdersGossipBatchSize {
+			break
+		}
+		orders = append(orders, order)
+		delete(n.ordersToGossip, orderHash)
+	}
+
+	now := time.Now().Unix()
+	selectedOrders := make([]*hubbleutils.SignedOrder, 0)
+	for _, order := range orders {
+		// skip gossip if the order is already expired
+		if order.ExpireAt.Int64() < now {
+			n.stats.IncSignedOrdersGossipOrderExpired()
+			continue
+		}
+		selectedOrders = append(selectedOrders, order)
+	}
+
+	if len(selectedOrders) == 0 {
+		return 0, nil
+	}
+
+	return len(selectedOrders), n.sendSignedOrders(selectedOrders)
+}
+
+func (n *pushGossiper) sendSignedOrders(orders []*hubbleutils.SignedOrder) error {
+	if len(orders) == 0 {
+		return nil
+	}
+
+	ordersBytes, err := rlp.EncodeToBytes(orders)
+	if err != nil {
+		return err
+	}
+	msg := message.SignedOrdersGossip{
+		Orders: ordersBytes,
+	}
+	msgBytes, err := message.BuildGossipMessage(n.codec, msg)
+	if err != nil {
+		return err
+	}
+
+	log.Trace(
+		"gossiping signed orders",
+		"len(orders)", len(orders),
+		"size(orders)", len(msg.Orders),
+	)
+	n.stats.IncSignedOrdersGossipSent(int64(len(orders)))
+	n.stats.IncSignedOrdersGossipBatchSent()
+	return n.client.Gossip(msgBytes)
+}
+
+//   #### HANDLER ####
+
+func (h *GossipHandler) HandleSignedOrders(nodeID ids.NodeID, msg message.SignedOrdersGossip) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	log.Trace(
+		"AppGossip called with SignedOrdersGossip",
+		"peerID", nodeID,
+		"bytes(orders)", len(msg.Orders),
+	)
+
+	if len(msg.Orders) == 0 {
+		log.Warn(
+			"AppGossip received empty SignedOrdersGossip Message",
+			"peerID", nodeID,
+		)
+		return nil
+	}
+
+	orders := make([]*hubbleutils.SignedOrder, 0)
+	if err := rlp.DecodeBytes(msg.Orders, &orders); err != nil {
+		log.Trace(
+			"AppGossip provided invalid orders",
+			"peerID", nodeID,
+			"err", err,
+		)
+		return nil
+	}
+
+	h.stats.IncSignedOrdersGossipReceived(int64(len(orders)))
+	h.stats.IncSignedOrdersGossipBatchReceived()
+
+	tradingAPI := h.vm.limitOrderProcesser.GetTradingAPI()
+	needToGossip := false
```

Comment:
Looks like this needs to be for every order.

Reply:
Yes, earlier I though that even if one order is new, then gossip the while batch. But I'll shift it to order level to gossip only new orders

Reply:
done